modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSourceTestCase.java

```diff
@@ -193,6 +200,33 @@ public void failToStart() throws Exception {
     messageSource.start();
   }
 
+  @Test
+  public void dispatchNotificationWhenFailToStart() throws Exception {
+    final Latch latch = new Latch();
+    final List<ExceptionNotification> notifications = new ArrayList<>();
+    final ExceptionNotificationListener listener = notification -> {
+      notifications.add(notification);
+      latch.release();
+    };
+    registerNotificationListener(listener);
+
+    ConnectionException connectionException = new ConnectionException(ERROR_MESSAGE);
+    MuleException e = new DefaultMuleException(connectionException);
+    doThrow(e).when(source).onStart(any());
+
+    messageSource.initialise();
+    try {
+      messageSource.start();
+      latch.await();
+      assertThat(notifications.get(0), is(instanceOf(ExceptionNotification.class)));
+      assertThat(notifications.get(0).getException(), instanceOf(RetryPolicyExhaustedException.class));
+    } catch (Exception ex) {
+      latch.await();
```

Comment:
who will release the latch if  start() failed?  If I'm wrong about this, add timeout

Reply:
If everything works the exception will be thrown after the the call to the system handler, so the latch will be released. I'm adding the timeout in case a future bug makes the exception be thrown before the call to the system handler, so the tests fails quickly